### PR TITLE
Excluding CLion IDE files with gitignore

### PR DESCRIPTION
**Pull request by Ririshi**
## Summary
Adds entries to `.gitignore` for the JetBrains CLion IDE, so I and others can use it without having to manually exclude the IDE files from our commits. Of course, this won't imply official support for the IDE, it is merely an exclusion of its files (like already exists for Visual Studio).

I could also provide others with the necessary files and settings to develop for ES in CLion. I may be able to write a setup guide if anyone would be interested in that. I would first have to figure out the specific settings for code reformatting in CLion that will be compatible with the ES codebase, though.

### CODE REVIEW

**Review by tehhowch — CHANGES_REQUESTED** [review]

File: .gitignore [hunk 1]
```diff
@@ -6,6 +6,11 @@
 # dlls needed to run the binary
 *.dll
 
+# Dependencies for compiling (available for download at
+# https://github.com/endless-sky/endless-sky/wiki/BuildInstructions#development-libraries)
+/dev64/
```
[the remark this answers is not in the document]
Comment (Ririshi) [reply .gitignore:11]: Actually the original gitignore already had the dev64 excluded, I just added the link and also excluded dev32.

File: .gitignore [hunk 2]
```diff
@@ -52,3 +57,7 @@ ipch/
 *.userosscache
 *.sln.docstates
 *.xcodeproj/**
+.idea/
+.idea_modules/
+cmake-build-*/
```
[the remark this answers is not in the document]
Comment (Ririshi) [reply .gitignore:62]: I could use any other folder, but CLion uses cmake-build-<build_type> by default. Could remove that line from the ignore, though, if you'd like, and I'll make sure to use an already-excluded folder.

**Review by Ririshi — COMMENTED** [review]
Fixed in next commit

**Comment by tehhowch** [comment]
Note that `x64` is an architectural build target, not the 64bit windows libraries

**Pull request merged**